### DISCUSSION TITLE
How do explicit transactions work with multiple entity managers?

### DISCUSSION THREAD

**Discussion by squivix** [mikro-orm/mikro-orm#4454] [Q&A] [answered]
Can I start a transaciton with one entity manager and commit or rollback with another? Does the data of an uncommited transaction from one entity manager show up in a query from another?

For example:

```ts
const em1=orm.em.fork();
await em1.begin()
await em1.create(User, {username:"johndoe"});

const em2=orm.em.fork();
const user=await em2.find(User, {username:"johndoe"})

//will this find a user or null?
console.log(user)

await em1.rollback();
```

I seem to get null, but I'm not sure if I'm doing something else wrong. Theoretically all queries are done to the same database with the same connection right?

**Comment by B4nan** [comment, marked as answer]
Nope, the EM instance is what holds transaction context, you cant mix those. 

> Theoretically all queries are done to the same database with the same connection right?

Nope, each fork can use a different connection, moreover, em.fork will give you clean fork, it does not inherit anything, in other words, it will have a new tx context. 

Also note that em.create is a sync method, you need to flush afterwards.
